composable_kernel/include/tensor_operation/gridwise_2d_reduction_blockwise.hpp

```diff
@@ -0,0 +1,863 @@
+                                                         OutDstVectorSize,
+                                                         1,
+                                                         false>(
+                            out1dDesc,
+                            make_multi_index(block_global_1d_id * M_BlockTileSize +
+                                             thread_dim0_cluster_id * MThreadSliceSize));
+
+                    threadwise_dst_load.Run(out1dDesc,
+                                            dst_global_buf,
+                                            ReducedDataDesc,
+                                            make_tuple(I0),
+                                            priorDstValue_buf);
+
+                    static_for<0, MThreadSliceSize, 1>{}([&](auto I) {
+                        accuValue_buf(I) += type_convert<AccDataType>(priorDstValue_buf[I] * beta);
+                    });
+                };
+            };
+
+            auto threadwise_dst_store =
+                ThreadwiseTensorSliceTransfer_v1r3<AccDataType,
+                                                   OutDataType,
+                                                   decltype(ReducedDataDesc),
+                                                   Out1dDescType,
+                                                   PassThroughOp<AccDataType>,
+                                                   Sequence<MThreadSliceSize>,
+                                                   Sequence<0>,
+                                                   0,
+                                                   OutDstVectorSize,
+                                                   InMemoryDataOperationEnum_t::Set,
+                                                   1,
+                                                   true>(
+                    out1dDesc,
+                    make_multi_index(block_global_1d_id * M_BlockTileSize +
+                                     thread_dim0_cluster_id * MThreadSliceSize),
+                    PassThroughOp<AccDataType>{});
+
+            threadwise_dst_store.Run(
+                ReducedDataDesc, make_tuple(I0), accuValue_buf, out1dDesc, dst_global_buf);
+        }
+    };
+
+    __device__ static void RunWithIndices(const In2dDescType& in2dDesc,
+                                          const Out1dDescType& out1dDesc,
+                                          const InElementwiseOperation& in_elementwise_op,
+                                          const OutElementwiseOperation& acc_elementwise_op,
+                                          AccDataType alpha,
+                                          const InDataType* const __restrict__ p_src_global,
+                                          OutDataType beta,
+                                          OutDataType* const __restrict__ p_dst_global,
+                                          const int* const __restrict__ ws_indices_global,
```

Comment:
same as above

index_t or pass the type as template arguement
ws_indices_global

Reply:
Agree!